### PR TITLE
Provide job schedules widget

### LINKED ISSUE #46: Provide a widget to define job schedules with
Which are the correct entries for JOB SCHEDULE field ?

## Expected Behavior
Suggest the correct values for this field

## Current Behavior
It is a free text filed: no choices are preconfigured

## Possible Solution
Please document what this field is needed for, including correct values to type in.
Or make availeble a choice list

## Steps to Reproduce (for bugs)
1. go to "create a now job"
2. in the right pane, click on the right of the word Schedule
3. write whatever you want
4. what is it needed for ??

## Your Environment
New Icinga installation - it is a lack of information, not a configuration specific bug.

IcingaX509Job

### PR DESCRIPTION
**Pull request by yhabteab**
resolves #46

### CODE REVIEW

**Comment by yhabteab** [comment]
### Requires

- `"dragonmantank/cron-expression": "^3.3.1"`

**Comment by nilmerg** [comment]
Things I noticed during testing the schedule element:

* You're still saving frequencies incorrectly here. (You're not calling `json_encode` yourself)
* Trying to update an existing job fails
  * The error message contains a typo: `Filed to update job...`

**Comment by yhabteab** [comment]
> Trying to update an existing job fails

Can you be a little more precise? This doesn't say anything about why the job update fails. I'm unable able to reproduce it.

**Comment by nilmerg** [comment]
Editing the end time of an existing job fails with the complaint *that a section "name" already exists*.

**Comment by yhabteab** [comment]
> Editing the end time of an existing job fails with the complaint _that a section "name" already exists_.

This can only be the case when you're trying to create a job with the same name, but never when editing it. Please check the urls in the search bar.

**Comment by yhabteab** [comment]
> Editing the end time of an existing job fails with the complaint _that a section "name" already exists_.

This is not only due to the new changes, but rather because the `INI` parser cannot handle **numeric** job names properly and is reproducible without this PR.
Bildschirm­foto 2023-02-27 um 13 42 48

File: application/forms/Config/JobConfigForm.php [hunk 1]
```diff
@@ -4,103 +4,173 @@
...
-                    'label'         => $this->translate('Schedule')
-                ]
-            ],
-        ]);
+        $this->identifier = Url::fromRequest()->getParam('name');
+        $this->scheduleElement = new ScheduleElement('schedule-element');
+        $this->scheduleElement->setIdProtector([Icinga::app()->getRequest(), 'protectId']);
+        if ($this->hasDefaultElementDecorator()) {
+            $this->scheduleElement->setDefaultElementDecorator($this->getDefaultElementDecorator());
+        }
```
Comment (nilmerg) [inline application/forms/Config/JobConfigForm.php:43]: Is this still required?

File: application/forms/Config/JobConfigForm.php [hunk 2]
```diff
@@ -4,103 +4,173 @@
...
+use function ipl\Stdlib\get_php_type;
 
 /**
  * Create, update and delete jobs
  */
-class JobConfigForm extends RepositoryForm
+class JobConfigForm extends CompatForm
 {
-    protected function createInsertElements(array $formData)
+    protected $identifier;
```
Comment (nilmerg) [inline application/forms/Config/JobConfigForm.php:28]: Please add a docstring.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: application/forms/Config/JobConfigForm.php (hunk 1 above)
Comment (yhabteab) [reply application/forms/Config/JobConfigForm.php:43]: Nope!

**Review by nilmerg — APPROVED** [review]

**Pull request merged**